coderd/workspacebuilds_test.go

```diff
@@ -1334,29 +1342,63 @@ func TestWorkspaceBuildTimings(t *testing.T) {
 			ResourceID: resource.ID,
 		})
 
-		// Then: return a response with empty agent script timings
+		// When: fetching timings for the build
 		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
 		t.Cleanup(cancel)
 		res, err := client.WorkspaceBuildTimings(ctx, build.ID)
 		require.NoError(t, err)
+
+		// Then: return a response with empty agent script timings
 		require.Empty(t, res.AgentScriptTimings)
 	})
 
 	// Some workspaces might not have agents. It is improbable, but possible.
 	//nolint:paralleltest
 	t.Run("NoAgents", func(t *testing.T) {
-		// When: fetching timings for a build with no agents
+		// Given: a build with no agents
 		build := makeBuild()
 		dbgen.WorkspaceResource(t, db, database.WorkspaceResource{
 			JobID: build.JobID,
 		})
 
-		// Then: return a response with empty agent script timings
-		// trigger build
+		// When: fetching timings for the build
 		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
 		t.Cleanup(cancel)
 		res, err := client.WorkspaceBuildTimings(ctx, build.ID)
 		require.NoError(t, err)
+
+		// Then: return a response with empty agent script timings
 		require.Empty(t, res.AgentScriptTimings)
 	})
+
+	//nolint:paralleltest
+	t.Run("AgentConnectionTimings", func(t *testing.T) {
+		// Given: a build with multiple agents
+		build := makeBuild()
+		resource := dbgen.WorkspaceResource(t, db, database.WorkspaceResource{
+			JobID: build.JobID,
+		})
+		agents := make([]database.WorkspaceAgent, 5)
```

Comment:
This is the default case, and 5 agents is not the common case; a single agent is the default, so we should represent that.